integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/ImplicitNestedSearchPredicateIT.java

```diff
@@ -0,0 +1,254 @@
+	}
+
+	@Test
+	public void predicate_simpleQueryString() {
+		verify_implicit_nest( p -> p.simpleQueryString().field( "nested.text" ).matching( SOME_SIMPLE_QUERY_STRING ) );
+	}
+
+	@Test
+	public void predicate_range() {
+		verify_implicit_nest( p -> p.range().field( "nested.numeric" ).atMost( ANY_INTEGER ) );
+	}
+
+	@Test
+	public void predicate_geoPoly() {
+		verify_implicit_nest( p -> p.spatial().within().field( "nested.geo" ).polygon( GeoPolygon.of( G00, G02, G22, G20, G00 ) ) );
+	}
+
+	@Test
+	public void predicate_geoBox() {
+		verify_implicit_nest( p -> p.spatial().within().field( "nested.geo" ).boundingBox( GeoBoundingBox.of( G20, G02 ) ) );
+	}
+
+	@Test
+	public void predicate_geoCircle() {
+		verify_implicit_nest( p -> p.spatial().within().field( "nested.geo" ).circle( G11, 1 ) );
+	}
+
+	@Test
+	public void predicate_exists_field() {
+		verify_implicit_nest( p -> p.exists().field( "nested.geo" ) );
+	}
+
+	@Test
+	public void predicate_exists_object() {
+		verify_implicit_nest( p -> p.exists().field( "nested" ) );
+	}
+
+	@Test
+	public void predicate_simpleQueryString_multipleNestedPaths() {
+		SubTest.expectException( () -> indexManager.createScope()
+				.predicate().simpleQueryString().field( "nested.text" ).field( "text" )
+		)
+				.assertThrown()
+				.isInstanceOf( SearchException.class )
+				.hasMessageContaining( "Simple query string targets fields" )
+				.hasMessageContaining( "spanning multiple nested paths" )
+				.hasMessageContaining( "nested.text" );
+	}
+
+	private void verify_implicit_nest(Function<? super SearchPredicateFactory, ? extends PredicateFinalStep> implicitPredicate) {
+		StubMappingScope scope = indexManager.createScope();
+		SearchPredicate explicitPredicate = scope.predicate().nested().objectField( "nested" ).nest( implicitPredicate ).toPredicate();
+
+		// test the explicit form
+		List<DocumentReference> docs = scope.query().selectEntityReference()
+				.where( explicitPredicate )
+				.fetchAllHits();
+
+		assertThat( docs ).hasDocRefHitsAnyOrder( INDEX_NAME, DOCUMENT_1 );
+
+		// test the implicit form
+		docs = scope.query().selectEntityReference()
+				.where( implicitPredicate )
+				.fetchAllHits();
+
+		assertThat( docs ).hasDocRefHitsAnyOrder( INDEX_NAME, DOCUMENT_1 );
+	}
+
+	private void initData() {
+		IndexIndexingPlan plan = indexManager.createIndexingPlan();
+		plan.add( referenceProvider( DOCUMENT_1 ), document -> {
+			DocumentElement nestedDocument = document.addObject( indexMapping.nested );
+			nestedDocument.addValue( indexMapping.nestedString, ANY_STRING );
+			nestedDocument.addValue( indexMapping.nestedNumeric, ANY_INTEGER );
+			nestedDocument.addValue( indexMapping.nestedText, SOME_PHRASE_TEXT );
+			nestedDocument.addValue( indexMapping.nestedGeo, G11 );
+
+			DocumentElement nestedDocumentX2 = nestedDocument.addObject( indexMapping.nestedX2 );
+			nestedDocumentX2.addValue( indexMapping.nestedX2Numeric, ANY_INTEGER );
+		} );
```

Comment:
You may want to index another document, just to be sure that predicates are actually taken into account... You would only expect document 1 to turn up in search hits.

Reply:
I added the document `DOCUMENT_2` that basically does not match any predicate, with the exception of `matchAll` and `exists` predicates. See the commit:

> HSEARCH-3752 Add a non-matching document to the test case